File: Arcade/Intro/24_minesweeper.js
function minesweeper(matrix) {
    var mines = [];
    
    // Get height and width of matrix
    var width = matrix[0].length;
    var height = matrix.length;
    
    for (var i = 0; i < height; i++) {
        // Reinitialize array as empty array
        var arr = [];
        for (var j = 0; j < width; j++) {
            // Reset count to zero
            var count = 0;
            for (var x = -1; x < 2; x++) {
                for (var y = -1; y < 2; y++) {
                    if (y === 0 && x === 0) {
                        // We don't want to count current cell
                    } else if ((i+x) >= 0 && (j+y) >= 0 && (i+x) < height && (j+y) < width) {
                        // Once we know cell is in bounds, we can check if it's true
                        count += matrix[i+x][j+y] === true ? 1 : 0;
                    }
                }
            }
            // Push count to the array
            arr.push(count);
        }
        // Push array to multidimensional array
        mines.push(arr);
    }
    
    return mines;
}

File: Arcade/Intro/C++/24_minesweeper.cpp
std::vector<std::vector<int>> minesweeper(std::vector<std::vector<bool>> matrix) {
    vector<vector<int>> result(matrix.size(),vector<int>(matrix[0].size(), 0));
    for(int i=0;i<matrix.size();i++){
        for(int j=0;j<matrix[0].size();j++){
            bool t=i>0;
            bool r=j<matrix[0].size()-1;
            bool b=i<matrix.size()-1;
            bool l=j>0;
            int sum=0;
            if(t && matrix[i-1][j]) sum++;
            if(t&&r&&matrix[i-1][j+1]) sum++;
            if(r&&matrix[i][j+1]) sum++;
            if(b&&r&&matrix[i+1][j+1]) sum++;
            if(b&&matrix[i+1][j]) sum++;
            if(b&&l&&matrix[i+1][j-1]) sum++;
            if(l&&matrix[i][j-1]) sum++;
            if(l&&t&&matrix[i-1][j-1]) sum++;
            
            result[i][j]=sum;
        }
    }
    return result;
}
